### PR TITLE
Issue #8198: Fixed beginning development instructions

### LINKED ISSUE #8198: Unclear beginning development instructions
https://checkstyle.org/beginning_development.html
The point number 6 given in this document results in a ton of commits being pulled from the main master and being pushed to the **current branch**. All these commits are visible and reflect in the PR, if already created.

I think we should make this addition to the 6th point:
```
git checkout master
git pull main master
git push origin master
```
Then point 7 becomes:
```
git checkout my-new-check
git rebase master
```
This will avoid confusion save a lot of time for someone who is new to contributing and git.

### PR DESCRIPTION
**Pull request by shashwat70**
Resolves #8198

### CODE REVIEW

**Comment by shashwat70** [comment]
> Thanks a lot for update.

It's my utmost pleasure!

> It would be awesome if you also rename 'main' to 'upstream' in this or separate PR.

Done in a separate commit. @romani 
Is it okay?

File: src/xdocs/beginning_development.xml [hunk 1]
```diff
@@ -53,7 +53,7 @@
       </p>
       <div class="wrap-content">
         <source>
-          git remote add main https://github.com/checkstyle/checkstyle
+          git remote add upstream https://github.com/checkstyle/checkstyle
```
Comment (romani) [inline src/xdocs/beginning_development.xml:56]: please make update at line `Fork your copy from Checkstyle main project. As it is described`
to `Fork Checkstyle upstream project. As it is described`

Comment (shashwat70) [reply src/xdocs/beginning_development.xml:56]: done @romani

File: src/xdocs/beginning_development.xml [hunk 2]
```diff
@@ -27,7 +27,7 @@
         You can find information about development environment preparation here:
         <a href="https://github.com/sevntu-checkstyle/sevntu.checkstyle/wiki/Prepare-Development-Environment-in-Ubuntu-12.04">
         Prepare development environment in Ubuntu</a>.<br/>
-        2. Fork your copy from Checkstyle main project. As it is described
+        2. Fork your copy from Checkstyle upstream project. As it is described
```
Comment (romani) [inline src/xdocs/beginning_development.xml:30]: please make it `Fork Checkstyle upstream project. As it is described`

Comment (shashwat70) [reply src/xdocs/beginning_development.xml]: I'm sorry I missed that. Done. @romani

**Pull request merged**